pkg/ebpf/c/tracee.bpf.c

```diff
@@ -1913,36 +1916,46 @@ static __always_inline int init_program_data(program_data_t *p, void *ctx)
     p->event->buf_off = 0;
 
     bool container_lookup_required = true;
+
     p->task_info = bpf_map_lookup_elem(&task_info_map, &p->event->context.task.host_tid);
     if (unlikely(p->task_info == NULL)) {
-        p->task_info =
-            init_task_info(p->event->context.task.host_tid, p->event->context.task.host_pid);
+        p->task_info = init_task_info(
+            p->event->context.task.host_tid,
+            p->event->context.task.host_pid,
+            kthread
+        );
         if (unlikely(p->task_info == NULL)) {
             return 0;
         }
-        // we just initialized task info so we know that recompute_scope is already set to true
+        // just initialized task info: recompute_scope is already set to true
         goto out;
     }
 
     // check if we need to recompute scope due to context change
     if (context_changed(&p->task_info->context, &p->event->context.task))
         p->task_info->recompute_scope = true;
 
-    // If task is already part of a container, there is no need to check if state changed
     u8 container_state = p->task_info->container_state;
-    if (container_state == CONTAINER_STARTED || container_state == CONTAINER_EXISTED) {
-        p->event->context.task.flags |= CONTAINER_STARTED_FLAG;
-        container_lookup_required = false;
+
+    // if task is already part of a container: no need to check if state changed
+    switch (container_state) {
```

Comment:
Are you sure that this is better?
I don't think we will have another case in the future, and I think switch is a bit less performant efficient than if-else.

Reply:
![image](https://user-images.githubusercontent.com/7395852/214092472-3e919d6f-033b-44f3-9312-4ed4b9455553.png)

same optimization for both cases, switch causes shorter lines and it is usually better for readability (based on my linters and the picture above).